File: Simplified/AppInfrastructure/NYPLConfiguration.m
#import "NYPLConfiguration.h"
#import "NYPLAppDelegate.h"

#import "UILabel+NYPLAppearanceAdditions.h"
#import "UIButton+NYPLAppearanceAdditions.h"
#import "SimplyE-Swift.h"

#if defined(FEATURE_DRM_CONNECTOR)
#import <ADEPT/ADEPT.h>
#endif

@implementation NYPLConfiguration

+ (NSURL *)mainFeedURL
{
  NSURL *const customURL = [NYPLSettings sharedSettings].customMainFeedURL;

  if(customURL) return customURL;

  NSURL *const accountURL = [NYPLSettings sharedSettings].accountMainFeedURL;
  return accountURL;
}

+ (NSURL *)minimumVersionURL
{
  return [NSURL URLWithString:@"http://www.librarysimplified.org/simplye-client/minimum-version"];
}

+ (UIColor *)accentColor
{
  return [UIColor colorWithRed:0.0/255.0 green:144/255.0 blue:196/255.0 alpha:1.0];
}

+ (UIColor *)readerBackgroundColor
{
  return [UIColor colorWithWhite:250/255.0 alpha:1.0];
}

// OK to leave as static color because it's reader-only
+ (UIColor *)readerBackgroundDarkColor
{
  return [UIColor colorWithWhite:5/255.0 alpha:1.0];
}

// OK to leave as static color because it's reader-only
+ (UIColor *)readerBackgroundSepiaColor
{
  return [UIColor colorWithRed:250/255.0 green:244/255.0 blue:232/255.0 alpha:1.0];
}

// OK to leave as static color because it's reader-only
+ (UIColor *)backgroundMediaOverlayHighlightColor
{
  return [UIColor yellowColor];
}

// OK to leave as static color because it's reader-only
+ (UIColor *)backgroundMediaOverlayHighlightDarkColor
{
  return [UIColor orangeColor];
}

// OK to leave as static color because it's reader-only
+ (UIColor *)backgroundMediaOverlayHighlightSepiaColor
{
  return [UIColor yellowColor];
}

// Set for the whole app via UIView+NYPLFontAdditions.
+ (NSString *)systemFontName
{
  return @"AvenirNext-Medium";
}

// Set for the whole app via UIView+NYPLFontAdditions.
+ (NSString *)boldSystemFontName
{
  return @"AvenirNext-Bold";
}

+ (NSString *)systemFontFamilyName
{
  return @"Avenir Next";
}

+ (CGFloat)defaultTOCRowHeight
{
  return 56;
}

+ (CGFloat)defaultBookmarkRowHeight
{
  return 100;
}

@end

File: Simplified/AppInfrastructure/NYPLConfiguration.swift
//
//  NYPLConfiguration.swift
//  Simplified
//
//  Created by Ettore Pasquini on 6/3/22.
//  Copyright © 2022 NYPL. All rights reserved.
//

import UIKit

class NYPLConfiguration: NSObject {
  private override init() {
    super.init()
  }

  // MARK: - Objects

  static var mainFeedURL: URL? {
    if let customURL = NYPLSettings.shared.customMainFeedURL {
      return customURL
    }

    return NYPLSettings.shared.accountMainFeedURL
  }

  // MARK: - Colors

  @objc static var accentColor: UIColor {
    return UIColor(red: 0.0/255.0, green: 144/255.0, blue: 196/255.0, alpha:1.0)
  }

  @objc static var readerBackgroundColor: UIColor {
    return UIColor(white: 250/255.0, alpha:1.0)
  }

  // OK to leave as static color because it's reader-only
  @objc static var readerBackgroundDarkColor: UIColor {
    return UIColor(white: 5/255.0, alpha:1.0)
  }

  // OK to leave as static color because it's reader-only
  @objc static var readerBackgroundSepiaColor: UIColor {
    return UIColor(red: 250/255.0, green: 244/255.0, blue: 232/255.0, alpha: 1.0)
  }

  // OK to leave as static color because it's reader-only
  @objc static var backgroundMediaOverlayHighlightColor: UIColor {
    return .yellow
  }

  // OK to leave as static color because it's reader-only
  @objc static var backgroundMediaOverlayHighlightDarkColor: UIColor {
    return .orange
  }

  // OK to leave as static color because it's reader-only
  @objc static var backgroundMediaOverlayHighlightSepiaColor: UIColor {
    return .yellow
  }

  // MARK: - Fonts

  // Set for the whole app via UIView+NYPLFontAdditions.
  @objc static var systemFontName: String {
    return "AvenirNext-Medium"
  }

  // Set for the whole app via UIView+NYPLFontAdditions.
  static var boldSystemFontName: String {
    return "AvenirNext-Bold"
  }

  @objc static var systemFontFamilyName: String {
    return "Avenir Next"
  }

  //MARK: - Dimensions

  static var defaultTOCRowHeight: CGFloat {
    return 56
  }

  static var defaultBookmarkRowHeight: CGFloat {
    return 100
  }

  static var cornerRadius: CGFloat {
    return 5
  }
}
